Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/DirectorySymlinkTests.cs

```diff
@@ -547,6 +547,32 @@ public void ConcurrentAccessToHardlinksPointingToSameFileIntertwinedWithOtherRea
             RunScheduler().AssertSuccess();
         }
 
+        [FactIfSupported(requiresSymlinkPermission: true)]
+        public void EnumerateDirectoryViaDirectorySymlinkShouldBeObservedAsDirectoryEnumeration()
+        {
+            AbsolutePath targetDirectory = CreateUniqueDirectory(ReadonlyRoot, "Target");
+            CreateSourceFile(targetDirectory, "file1");
+            CreateSourceFile(targetDirectory, "file2");
+
+            AbsolutePath directorySymlink = CreateUniquePath("Symlink", ReadonlyRoot);
+            XAssert.IsTrue(FileUtilities.TryCreateSymbolicLink(
+                directorySymlink.ToString(Context.PathTable), 
+                targetDirectory.ToString(Context.PathTable), 
+                isTargetFile: false).Succeeded);
+
+            ProcessWithOutputs processWithOutputs = CreateAndSchedulePipBuilder(new[]
+            {
+                Operation.EnumerateDir(DirectoryArtifact.CreateWithZeroPartialSealId(directorySymlink)),
+                Operation.WriteFile(CreateOutputFileArtifact())
+            });
+
+            RunScheduler().AssertSuccess();

```

Comment:
On macOS you have to allow access to `directorySymlink` file, otherwise you'll get:
```
[0:04] Error DX0500: [PipEB9A5D1F51C0C26E, Test.BuildXL.Executables.TestProcess, TestModule, testValue, {}] - Disallowed file accesses were detected (R = read, W = write):
        Disallowed file accesses performed by: /usr/bin/time
         R .../TestProj/Out/Objects.noindex/a/x/o3qjkq7e5t4nqzoz8wqko3e1g64rte/t_10/DirectorySym4A8FD4B8/0/obj/readonly/Symlink_3
```

This should do the trick:
```sharp
var pipBuilder = CreatePipBuilder(new[]
            {
                Operation.EnumerateDir(DirectoryArtifact.CreateWithZeroPartialSealId(directorySymlink)),
                Operation.WriteFile(CreateOutputFileArtifact())
            });
            pipBuilder.AddInputFile(directorySymlink);
            ProcessWithOutputs processWithOutputs = SchedulePipBuilder(pipBuilder);
```

Reply:
Thanks for testing it on Mac. I was relying on the PR build, but the first time it builds, some weird things happened.

---
In reply to: [283882342](https://github.com/microsoft/BuildXL/pull/301#discussion_r283882342) [](ancestors = 283882342)